File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerTopProductResponse.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.order.entity.SellerTopProduct;

import java.math.BigDecimal;
import java.util.UUID;

public record SellerTopProductResponse(UUID sellerId,
                                       UUID productId,
                                       String productName,
                                       Integer productPrice,
                                       String productImageUrl,
                                       Long orderCount,
                                       Long totalQuantitySold,
                                       BigDecimal totalSalesAmount,
                                       Integer salesRank) {
    public static SellerTopProductResponse fromEntity(SellerTopProduct sellerTopProduct) {
        return new SellerTopProductResponse(
                sellerTopProduct.getId().getSellerId(),
                sellerTopProduct.getId().getProductId(),
                sellerTopProduct.getProductName(),
                sellerTopProduct.getProductPrice().intValue(),
                sellerTopProduct.getProductImageUrl(),
                sellerTopProduct.getOrderCount(),
                sellerTopProduct.getTotalQuantitySold(),
                sellerTopProduct.getTotalSalesAmount(),
                sellerTopProduct.getSalesRank()
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerTopProductResponse.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto

import java.math.BigDecimal
import java.util.UUID

data class SellerTopProductResponse(
    val sellerId: UUID,
    val productId: UUID,
    val productName: String,
    val productPrice: Int,
    val productImageUrl: String?,
    val orderCount: Long,
    val totalQuantitySold: Long,
    val totalSalesAmount: BigDecimal,
    val salesRank: Int
)
